src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/store/utils/build_workflow_lookup.ts

```diff
@@ -104,6 +111,15 @@ function inspectStep(node: any, lineCounter: LineCounter): Record<string, StepIn
   }
 
   if (stepId && stepType) {
+    const stepNode = node as YAML.YAMLMap<unknown, unknown>;
```

Comment:
Could we do `if (YAML.isMap(node)) {` instead of casting? From the code above, it seems the node is not always a map

Reply:
+1, please!